addon/components/pie-chart.js

```diff
@@ -483,9 +483,17 @@ export default ChartComponent.extend(FloatingTooltipMixin,
     groups.select('path').attr(this.get('sliceAttrs'));
     
     var labelWidth = this.get('labelWidth');
+    var maxLabelWidth = this.get('outerWidth')/2 - this.get('labelPadding');
     var labelTrimmer = LabelTrimmer.create({
-      getLabelSize: function() {
-        return labelWidth;
+      getLabelSize: function(d, selection) {
+        var text, xPos;
+        // Pick the current text element by filtering on data
+        text = selection.filter(function(data) {
+          return data === d;
+        });
+        // Calculate current horizontal translation from the center
+        xPos = d3.transform(text.attr("transform")).translate[0];
+        return Math.max(labelWidth, maxLabelWidth - Math.abs(xPos));
```

Comment:
This would enforce that all labels' widths are at least 1/4 of the width of the chart. I don't feel like that's desirable behavior, although it might be nice from a backwards compatibility standpoint.